precompiles/identity/src/lib.rs

```diff
@@ -0,0 +1,845 @@
+
+		let origin = Runtime::AddressMapping::into_account_id(caller);
+		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		event.record(handle)?;
+
+		Ok(())
+	}
+
+	#[precompile::public("quitSub()")]
+	fn quit_sub(handle: &mut impl PrecompileHandle) -> EvmResult {
+		let caller = handle.context().caller;
+
+		let event = log1(
+			handle.context().address,
+			SELECTOR_LOG_SUB_IDENTITY_REVOKED,
+			solidity::encode_event_data(Address(caller)),
+		);
+		handle.record_log_costs(&[&event])?;
+
+		let call = pallet_identity::Call::<Runtime>::quit_sub {};
+
+		let origin = Runtime::AddressMapping::into_account_id(caller);
+		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		event.record(handle)?;
+
+		Ok(())
+	}
+
+	#[precompile::public("identity(address)")]
+	#[precompile::view]
+	fn identity(
+		handle: &mut impl PrecompileHandle,
+		who: Address,
+	) -> EvmResult<Registration<Runtime::MaxAdditionalFields>> {
+		// Storage item: IdentityOf:
+		// BoundedVec((RegistrarIndex(4) + Judgement(17)) * MaxRegistrars(20)) + Balance(16)
+		//        + IdentityInfo(BoundedVec(Data(33) * MaxAdditionalFields(100)) + Data(33)*7 + 20)
+		handle.record_db_read::<Runtime>(3987)?;
+
+		let who: H160 = who.into();
+		let who = Runtime::AddressMapping::into_account_id(who);
+		let identity = pallet_identity::Pallet::<Runtime>::identity(who);
+
+		Ok(Self::identity_to_output(identity)?)
+	}
+
+	#[precompile::public("superOf(address)")]
+	#[precompile::view]
+	fn super_of(handle: &mut impl PrecompileHandle, who: Address) -> EvmResult<SuperOf> {
+		// Storage item: SuperOf:
+		// AccountId(20) + Data(33)
+		handle.record_db_read::<Runtime>(53)?;
+
+		let who: H160 = who.into();
+		let who = Runtime::AddressMapping::into_account_id(who);
+		if let Some((account, data)) = pallet_identity::Pallet::<Runtime>::super_of(who) {
+			Ok(SuperOf {
+				is_valid: true,
+				account: Address(account.into()),
+				data: Self::data_to_output(data),
+			})
+		} else {
+			Ok(SuperOf::default())
+		}
+	}
+
+	#[precompile::public("subsOf(address)")]
+	#[precompile::view]
+	fn subs_of(handle: &mut impl PrecompileHandle, who: Address) -> EvmResult<SubsOf> {
+		// Storage item: SubsOf:
+		// BoundedVec(AccountId(20) * MaxSubAccounts(100))
+		handle.record_db_read::<Runtime>(2000)?;
+
+		let who: H160 = who.into();
+		let who = Runtime::AddressMapping::into_account_id(who);
+		let (deposit, accounts) = pallet_identity::Pallet::<Runtime>::subs_of(who);
+
+		let accounts = accounts
+			.into_iter()
+			.map(|account| Address(account.into()))
+			.collect();
+
+		Ok(SubsOf {
+			deposit: deposit.into(),
+			accounts,
+		})
+	}
+
+	#[precompile::public("registrars()")]
+	#[precompile::view]
+	fn registrars(handle: &mut impl PrecompileHandle) -> EvmResult<Vec<Registrar>> {
+		// Storage item: Registrars:
+		// BoundedVec((AccountId(20) + Balance(16) + IdentityFields(8)) * MaxSubAccounts(100))
+		handle.record_db_read::<Runtime>(4400)?;
```

Comment:
Ideally we would not hard code `100` here. Also, could we use `RegistrarInfo::max_encoded_len()`?